src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -4299,6 +4230,99 @@ private bool HasTopLevelNullabilityConversion(TypeWithAnnotations source, TypeWi
             return resultType;
         }
 
+        private TypeWithState ApplyUserDefinedConversion(
+            BoundExpression node,
+            BoundExpression operandOpt,
+            Conversion conversion,
+            TypeWithAnnotations targetTypeWithNullability,
+            TypeWithState operandType,
+            bool useLegacyWarnings,
+            AssignmentKind assignmentKind,
+            ParameterSymbol target,
+            bool reportTopLevelWarnings,
+            bool reportRemainingWarnings)
+        {
+            Debug.Assert(!IsConditionalState);
+            Debug.Assert(node != null);
+            Debug.Assert(operandOpt != null || !operandType.HasNullType);
+            Debug.Assert(targetTypeWithNullability.HasType);
+            Debug.Assert((object)target != null || assignmentKind != AssignmentKind.Argument);
+
+            TypeSymbol targetType = targetTypeWithNullability.Type;
+            Debug.Assert(conversion.Kind == ConversionKind.ExplicitUserDefined || conversion.Kind == ConversionKind.ImplicitUserDefined);
```

Comment:
nit: group this with the other asserts at the top of the method?

Reply:
Will move in next PR.

---
In reply to: [269783790](https://github.com/dotnet/roslyn/pull/34496#discussion_r269783790) [](ancestors = 269783790)